src/Compatibility/Core/src/Android/PopupManager.cs

```diff
@@ -50,20 +51,17 @@ internal sealed class PopupRequestHelper : IDisposable
 			internal PopupRequestHelper(Activity context)
 			{
 				Activity = context;
-				MessagingCenter.Subscribe<Page, bool>(Activity, Page.BusySetSignalName, OnPageBusy);
-				MessagingCenter.Subscribe<Page, AlertArguments>(Activity, Page.AlertSignalName, OnAlertRequested);
-				MessagingCenter.Subscribe<Page, PromptArguments>(Activity, Page.PromptSignalName, OnPromptRequested);
-				MessagingCenter.Subscribe<Page, ActionSheetArguments>(Activity, Page.ActionSheetSignalName, OnActionSheetRequested);
+
+				WeakReferenceMessenger.Default.Register<Activity, PageBusyMessage>(Activity, (r, m) => OnPageBusy(m.Page, m.IsBusy));
+				WeakReferenceMessenger.Default.Register<Activity, PageAlertMessage>(Activity, (r, m) => OnAlertRequested(m.Page, m.Arguments));
+				WeakReferenceMessenger.Default.Register<Activity, PromptMessage>(Activity, (r, m) => OnPromptRequested(m.Page, m.Arguments));
+				WeakReferenceMessenger.Default.Register<Activity, ActionSheetMessage>(Activity, (r, m) => OnActionSheetRequested(m.Page, m.Arguments));
```

Comment:
Nit: if we don't care about the strongly-typed recipient (it's not used in the handler), this can be simplified.
Additionally (though up to style preference), the recipient can be a discard to make it more explicit:

```suggestion
				WeakReferenceMessenger.Default.Register<PageBusyMessage>(Activity, (_, m) => OnPageBusy(m.Page, m.IsBusy));
				WeakReferenceMessenger.Default.Register<PageAlertMessage>(Activity, (_, m) => OnAlertRequested(m.Page, m.Arguments));
				WeakReferenceMessenger.Default.Register<PromptMessage>(Activity, (_, m) => OnPromptRequested(m.Page, m.Arguments));
				WeakReferenceMessenger.Default.Register<ActionSheetMessage>(Activity, (_, m) => OnActionSheetRequested(m.Page, m.Arguments));
```